core/validation/validation.go

```diff
@@ -20,7 +20,11 @@ import (
 	"github.com/vmware-tanzu/secrets-manager/core/env"
 )
 
-const spiffeRegexPrefixStart = "^"
+// Any SPIFFE ID regular expression matcher shall start with the
+// `^spiffe://$trustDomain` prefix for extra security.
+//
+// This variable shall be treated as constant and should not be modified.
+var spiffeRegexPrefixStart = "^spiffe://" + env.SpiffeTrustDomain() + "/"
```

Comment:
This is intentionally hard coded and kept reconfigurable because

1. Any spiffe id has to start with `spiffe://` as part of the standard and then followed by the trust domain.
2. Starting the matcher with a `^` makes it more restrictive and more secure, since you cannot just use any regexp, and you have to match thing starting from the beginning of the string.